### PR TITLE
Validating placements tests

### LINKED ISSUE #11: Board #valid_placement? test

### LINKED ISSUE #13: Board #place test

### LINKED ISSUE #14: Board #render test

### LINKED ISSUE #18: Board #valid_coordinate? test

### PR DESCRIPTION
**Pull request by kbergstrom78**
Added remaining tests for iteration 2.

### CODE REVIEW

File: spec/board_spec.rb [hunk 1]
```diff
@@ -32,4 +32,61 @@
       expect(@board.valid_coordinate?("A22")).to be  false
     end
   end
+
+  describe 'placment validation' do
+    it 'checks before ship placement' do
```
Comment (kaileykaes) [inline spec/board_spec.rb:37]: We need another line of set-up define the cruiser variable. This is from the IP prior to the source of these expect statements:
`cruiser = Ship.new("Cruiser", 3)`
`submarine = Ship.new("Submarine", 2)`
We can either instantiate these on lines 8.5 and 9 in the `before(:each)` statement thusly
`@cruiser = Ship.new("Cruiser", 3)` 
`@submarine = Ship.new("Submarine", 2)`
and refer to them in future tests by using the @ symbol before the variable name in subsequent tests, or we can instantiate them in each `it` block without the @ symbol.

File: spec/board_spec.rb [hunk 2]
```diff
@@ -32,4 +32,61 @@
...
+      expect(board.valid_placement?(cruiser, ["A1", "A2"])).to be false
+      expect(board.valid_placement?(submarine, ["A2", "A3", "A4"])). to be false
+    end
+  
+    it 'has consecutive coordinates' do
```
Comment (kaileykaes) [inline spec/board_spec.rb:42]: `cruiser` and `submarine` set-up here or with `@` symbol in the `before(:each)` statement

File: spec/board_spec.rb [hunk 3]
```diff
@@ -32,4 +32,61 @@
...
+      expect(board.valid_placement?(cruiser, ["A1", "A2", "A4"])).to be false
+      expect(board.valid_placement?(submarine, ["A1", "C1"])).to be false
+      expect(board.valid_placement?(cruiser, ["A3", "A2", "A1"])).to be false
+      expect(board.valid_placement?(submarine, ["C1", "B1"])).to be false
+    end
+  
+    it 'is not diagonal' do
```
Comment (kaileykaes) [inline spec/board_spec.rb:49]: `cruiser` and `submarine` set-up here or with `@` symbol in the `before(:each)` statement

File: spec/board_spec.rb [hunk 4]
```diff
@@ -32,4 +32,61 @@
...
+      expect(board.valid_placement?(cruiser, ["A1", "B2", "C3"])).to be false
+      expect(board.valid_placement?(submarine, ["C2", "D3"])).to be false
+    end
+
+    it '#valid_placement' do
```
Comment (kaileykaes) [inline spec/board_spec.rb:54]: `cruiser` and `submarine` set-up here or with `@` symbol in the `before(:each)` statement

File: spec/board_spec.rb [hunk 5]
```diff
@@ -32,4 +32,61 @@
...
+      expect(board.valid_placement?(submarine, ["A1", "A2"])).to be true
+      expect(board.valid_placement?(cruiser, ["B1", "C1", "D1"])).to be true
+    end
+  end
+
+  describe 'places ships' do 
+    it 'places cruiser' do
+      cruiser = Ship.new("Cruiser", 3)
```
Comment (kaileykaes) [inline spec/board_spec.rb:62]: Perfect set-up. You nailed all of the set-up hereafter. Nice.

File: spec/board_spec.rb (hunk 1 above)
Comment (kbergstrom78) [reply spec/board_spec.rb:37]: Oh shoot - good catch.  I thought I had caught all of those.  I will add it in now.

File: spec/board_spec.rb (hunk 2 above)
Comment (kbergstrom78) [reply spec/board_spec.rb:42]: Got it - will add.

**Pull request merged**